syne_tune/optimizer/schedulers/searchers/bayesopt/gpautograd/kernel/base.py

```diff
@@ -133,27 +133,24 @@ def _inverse_bandwidths(self):
     def forward(self, X1, X2):
         """Computes matrix of squared distances
 
-        :param X1: input matrix, shape ``(n1,d)``
-        :param X2: input matrix, shape ``(n2,d)``
+        :param X1: input matrix, shape ``(n1, d)``
+        :param X2: input matrix, shape ``(n2, d)``
         """
         # In case inverse_bandwidths if of size (1, dimension), dimension>1,
         # ARD is handled by broadcasting
         inverse_bandwidths = anp.reshape(self._inverse_bandwidths(), (1, -1))
 
+        X1_scaled = anp.multiply(X1, inverse_bandwidths)
+        X1_squared_norm = anp.sum(anp.square(X1_scaled), axis=1)
         if X2 is X1:
-            X1_scaled = anp.multiply(X1, inverse_bandwidths)
             D = -2.0 * anp.dot(X1_scaled, anp.transpose(X1_scaled))
-            X1_squared_norm = anp.sum(anp.square(X1_scaled), axis=1)
-            D = D + anp.reshape(X1_squared_norm, (1, -1))
-            D = D + anp.reshape(X1_squared_norm, (-1, 1))
+            X2_squared_norm = X1_squared_norm
```

Comment:
This line and the removal of the old line 148, are the only things that still confuse me in the PR. 

Do we really not need the old line 148 any more? It's handled by new lines 147, 152, 153?

Reply:
148 is now 153.